keps/sig-node/4603-tune-crashloopbackoff/README.md

```diff
@@ -0,0 +1,1794 @@
+the most conservative modeled initial value, 1s, resulting in 3 excess restarts
+per backoff lifetime. (See [this section](#front-loaded-decay-curve-methodology)
+for the source.]
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes, disable is supported.
+
+For `ReduceDefaultCrashLoopBackoffDecay`, if this is disabled, once kubelet is
+restarted it will initialize the default backoff to the prior initial value of
+10s, and all restart delays thereafter will be calculated against this equation.
+
+For `EnableRapidCrashLoopBackoffDecay`, if this is disabled, once kube-apiserver
+is restarted it will serve `restartPolicy` fields set to `Rapid` as `Always`.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Both features can also be reenabled.
+
+For `ReduceDefaultCrashLoopBackoffDecay`, if this is reenabled, once kubelet is
+restarted it will initialize the default backoff again to the new initial value
+of 1s, and all restart delays thereafter will be calculated against this
+equation.
+
+For `EnableRapidCrashLoopBackoffDecay`, if this is disabled, once kube-apiserver
+is restarted it will serve `restartPolicy` fields set to `Rapid` again as
+`Rapid`, which kubelet will be able to interpret.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Yes, this requires tests for
+* switching `ReduceDefaultCrashLoopBackoffDecay` on or off
+* switching `EnableRapidCrashLoopBackoffDecay` off when there are workloads with
+  `restartPolicy: Rapid` set
+* switching `EnableRapidCrashLoopBackoffDecay` off when there are no workloads
+  with `restartPolicy: Rapid` set
+* switching `EnableRapidCrashLoopBackoffDecay` off, and then back on again, when
+  there are workloads with `restartPolicy: Rapid` set
```

Comment:
We don't have good testing infrastructure for this. I'm not sure that all these cases are necessary. I don't think `ReduceDefaultCrashLoopBackoffDecay` needs enable/disable tests.

For `EnableRapidCrashLoopBackoffDecay` manual testing might be sufficient.